backend/core/src/main/java/org/sonarsource/sonarlint/core/ConnectionSuggestionProvider.java

```diff
@@ -0,0 +1,138 @@
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+import org.sonarsource.sonarlint.core.commons.progress.ExecutorServiceShutdownWatchable;
+import org.sonarsource.sonarlint.core.commons.progress.SonarLintCancelMonitor;
+import org.sonarsource.sonarlint.core.fs.ClientFile;
+import org.sonarsource.sonarlint.core.fs.FileSystemUpdatedEvent;
+import org.sonarsource.sonarlint.core.repository.config.ConfigurationRepository;
+import org.sonarsource.sonarlint.core.repository.connection.ConnectionConfigurationRepository;
+import org.sonarsource.sonarlint.core.rpc.protocol.SonarLintRpcClient;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.connection.ConnectionSuggestionDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.client.connection.SuggestConnectionParams;
+import org.springframework.context.event.EventListener;
+
+import static org.sonarsource.sonarlint.core.BindingClueProvider.ALL_BINDING_CLUE_FILENAMES;
+
+@Named
+@Singleton
+public class ConnectionSuggestionProvider {
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+
+  private final ConfigurationRepository configRepository;
+  private final ConnectionConfigurationRepository connectionRepository;
+  private final SonarLintRpcClient client;
+  private final BindingClueProvider bindingClueProvider;
+  private final ExecutorServiceShutdownWatchable<?> executorService;
+  private final BindingSuggestionProvider bindingSuggestionProvider;
+
+  @Inject
+  public ConnectionSuggestionProvider(ConfigurationRepository configRepository, ConnectionConfigurationRepository connectionRepository, SonarLintRpcClient client,
+                                   BindingClueProvider bindingClueProvider, BindingSuggestionProvider bindingSuggestionProvider) {
+    this.configRepository = configRepository;
+    this.connectionRepository = connectionRepository;
+    this.client = client;
+    this.bindingClueProvider = bindingClueProvider;
+    this.executorService = new ExecutorServiceShutdownWatchable<>(new ThreadPoolExecutor(0, 1, 10L, TimeUnit.SECONDS,
+      new LinkedBlockingQueue<>(), r -> new Thread(r, "Connection Suggestion Provider")));
+    this.bindingSuggestionProvider = bindingSuggestionProvider;
+  }
+
+  @EventListener
+  public void filesystemUpdated(FileSystemUpdatedEvent event) {
+    var listConfigScopeIds = event.getAddedOrUpdated().stream()
+      .filter(f -> ALL_BINDING_CLUE_FILENAMES.contains(f.getFileName()) || f.isSonarlintConfigurationFile())
+      .map(ClientFile::getConfigScopeId)
+      .collect(Collectors.toSet());
+
+    if (!listConfigScopeIds.isEmpty()) {
+      var cancelMonitor = new SonarLintCancelMonitor();
+      cancelMonitor.watchForShutdown(executorService);
+      executorService.submit(() -> suggestConnectionForGivenScopes(listConfigScopeIds, cancelMonitor));
+    }
+  }
+
+  private void suggestConnectionForGivenScopes(Set<String> listOfFilesPerConfigScopeIds, SonarLintCancelMonitor cancelMonitor) {
```

Comment:
The only occasion where we call this method is when the filesystem is updated. Shouldn't we have other triggers? For example, if a scope is added?